--- gg2-2.0.2/src/plugins/gadu_gadu/gadu_gadu_plugin.c~	2004-04-20 23:59:24.000000000 +0200
+++ gg2-2.0.2/src/plugins/gadu_gadu/gadu_gadu_plugin.c	2004-05-04 10:16:50.264475968 +0200
@@ -1494,9 +1494,9 @@
 	ggadu_config_var_add(handler, "ignore_unknown_sender", VAR_BOOL);
 	
 	if (g_getenv("HOME_ETC"))
-		this_configdir = g_build_filename(g_getenv("HOME_ETC"), "gg", NULL);
+		this_configdir = g_build_filename(g_getenv("HOME_ETC"), "gg2", NULL);
 	else
-		this_configdir = g_build_filename(g_get_home_dir(), ".gg", NULL);
+		this_configdir = g_build_filename(g_get_home_dir(), ".gg2", NULL);
 /* reserved for DEVEL branch 
 #if GGADU_DEBUG
     path = g_build_filename (this_configdir, "config_test", NULL);
